src/components/DownloadListButton.vue

```diff
@@ -0,0 +1,126 @@
+            calculatorStore: useCalculatorStore(),
+            ArrowRight,
+            items: [
+                {
+                    label: 'Download as JSON',
+                    command: () => {
+                        this.downloadAsJson()
+                    }
+                },
+                {
+                    label: 'Download as Navigator Layer',
+                    command: () => {
+                        this.downloadAsNavigatorLayer()
+                    }
+                }
+            ]
+        };
+    },
+    methods: {
+        downloadAsJson() {
+            const parsedList = [] as Array<ExportedTechnique>;
+
+            this.rankedList.slice(0, 10).forEach((technique, i) => {
+                const t = {
+                    rank: i + 1,
+                    tid: technique.tid,
+                    name: technique.name,
+                    description: technique.description,
+                    url: technique.url,
+                    detection: technique.detection,
+                    score: technique.adjusted_score,
+                    network_score: technique.network_score,
+                    process_score: technique.process_score,
+                    file_score: technique.file_score,
+                    cloud_score: technique.cloud_score,
+                    hardware_score: technique.hardware_score,
+                    mitigations: technique.mitigations,
+                    subtechniques: technique.subtechniques,
+                    actionability_score: technique.actionability_score,
+                    choke_point_score: technique.choke_point_score,
+                    prevalence_score: technique.prevalence_score,
+                }
+                parsedList.push(t);
+            })
+            downloadjs(JSON.stringify(parsedList, null, 4), "TopTenTechniques.json", JSON)
+        },
+        downloadAsNavigatorLayer() {
+            const gradient = ["#FFDDBD", "#ffaf66"]
+            const layer = {
+                "name": "Top 10 ATT&CK Techniques",
+                "versions": {
+                    "navigator": "4.8.0",
+                    "layer": "4.5",
+                    "attack": 10.1,
+                },
+                "sorting": 3,
+                "description": "Top ATT&CK Techniques heatmap overview of ATT&CK",
+                "domain": "enterprise-attack",
+                "techniques": [] as Array<{
+                    techniqueID: string;
+                    score: number;
+                    comment: string;
+                    metadata: never[];
+                }>,
+                "gradient": {
+                    "colors": gradient,
+                    "minValue": 0,
+                    "maxValue": 3
+                },
+            }
+            this.rankedList.forEach((technique, i) => {
+                let description = ` Rank: ${i + 1}`;
+                for (const monitoringType of Object.keys(this.calculatorStore.systemScore)) {
```

Comment:
So, currently, this implementation makes TypeScript angry. `Object.keys()` returns a `string[]`. That means `monitoringType` is a `string` and THAT means that the type of `${monitoringType}_score` is effectively `<string>_score`. This will cause problems when you try to reference:

```ts
technique[`${monitoringType}_score`]
```

Because TypeScript can't guarantee you'll resolve a valid `*_score` key in the `Technique` type. If you end up adding a new key to `systemScore` without adding an equivalent `*_score` key to `Technique`, this code will break and TypeScript won't be able to tell you.

We need to indicate to TypeScript that `Object.keys()` WON'T return a generic `string[]` in this case, but an array of the five keys defined by `systemScore`.

To do that, we'll want to export our CalculatorStore's type in `calculator.store.ts`:
```ts
...
      this.filterPropertiesObj.nist.options = nist_array;
      this.filterPropertiesObj.cis.options = cis_array;
      this.filterPropertiesObj.os.options = platform_array;
    },
  },
});

// Define Calculator Store Type
export type CalculatorStore = ReturnType<typeof useCalculatorStore>;
```

Then, we'll want to use that type in `DownloadListButton.vue` to indicate that `Object.keys()` won't return a generic `string[]` but an array of keys from the `systemScore` type:

```ts
type SystemScoreKeys = (keyof CalculatorStore["systemScoreObj"])[];
for (const monitoringType of Object.keys(this.calculatorStore.systemScore) as SystemScoreKeys) {
    description += ...
}
```
Now TypeScript will be happy :)

*Why doesn't `Object.keys()` work like this naturally? Why do I have to cast?*

Because new keys can be added to any object at runtime, so TypeScript can't be sure that `Object.keys()` will only return the keys defined by an object's TypeScript type. This doesn't apply in our case because we should never be adding keys at runtime that TypeScript doesn't know about.

Reply:
This is really helpful! Thank you!!